ql/src/java/org/apache/hadoop/hive/ql/io/orc/FixAcidKeyIndex.java

```diff
@@ -180,7 +180,8 @@ public static AcidKeyIndexValidationResult validate(Configuration conf, Path inp
         RecordIdentifier recordIdentifier = new RecordIdentifier(lastTransaction, lastBucket, lastRowId);
         result.recordIdentifiers.add(recordIdentifier);
 
-        if (stripes.size() != keyIndex.length || keyIndex[i] == null || recordIdentifier.compareTo(keyIndex[i]) != 0) {
+        if (keyIndex == null || stripes.size() != keyIndex.length || keyIndex[i] == null
```

Comment:
Since keyIndex is independent of the stripes,  the `if (keyIndex == null)` check can be done right after line 159  and the `result.isValid` set to false right then.  Otherwise, here we will be setting the flag repeatedly. The check here could be modified to 
`if (result.isValid && (stipes.size() ...<original condition>))`